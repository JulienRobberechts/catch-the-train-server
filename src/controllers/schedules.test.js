const SchedulesController = require("./schedules");
const {
  schedulesRepositoryMocked,
  missionsRepositoryMocked,
} = require("../adapters/ratp-api");
describe("schedules controller", () => {
  let controller;
  beforeAll(() => {
    controller = new SchedulesController({
      schedulesRepository: schedulesRepositoryMocked,
      missionsRepository: missionsRepositoryMocked,
    });
  });

  it("should not accept null parameter", async () => {
    const invalidControllerInit = () => new SchedulesController();
    expect(invalidControllerInit).toThrow();
  });
  it("should not accept empty parameter", async () => {
    const invalidControllerInit = () => new SchedulesController({});
    expect(invalidControllerInit).toThrow();
  });
  it("should not accept null api schedulesRepository", async () => {
    const invalidControllerInit = () =>
      new SchedulesController({
        schedulesRepository: null,
        missionsRepository: {},
      });
    expect(invalidControllerInit).toThrow();
  });
  it("should not accept null api missionsRepository", async () => {
    const invalidControllerInit = () =>
      new SchedulesController({
        schedulesRepository: {},
        missionsRepository: null,
      });
    expect(invalidControllerInit).toThrow();
  });
  it("should return schedules", async () => {
    const network = "rers";
    const line = "A";
    const fromStationSlug = "chatelet+les+halles";
    const toStationSlug = "maisons+laffitte";
    const result = await controller.getSchedulesForJourneyByDestination(
      network,
      line,
      fromStationSlug,
      toStationSlug
    );
    expect(result).toMatchSnapshot();
  });
});
