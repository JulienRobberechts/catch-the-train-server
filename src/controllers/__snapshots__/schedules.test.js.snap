// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schedules controller should return schedules 1`] = `
Object {
  "context": Object {
    "at": "2020-03-10T09:22:30+01:00",
    "line": "A",
    "missions": Array [
      "UPAC",
      "TEDI",
    ],
    "network": "rers",
    "provider": "ratp",
    "station": Object {
      "name": "Chatelet-Les-Halles",
      "slug": "chatelet+les+halles",
    },
  },
  "departures": Array [
    Object {
      "departureTime": "2020-03-10T09:24:00+01:00",
      "displayAttributes": "09:24",
      "displayDestination": "Cergy-Le-Haut",
      "isApproaching": undefined,
      "isDeparture": undefined,
      "isOnPlatform": undefined,
      "isTerminus": undefined,
      "mission": "UPAC",
      "noPassenger": undefined,
      "platform": undefined,
      "trainCode": "0924",
    },
    Object {
      "departureTime": "2020-03-10T09:29:00+01:00",
      "displayAttributes": "09:29",
      "displayDestination": "Cergy-Le-Haut",
      "isApproaching": undefined,
      "isDeparture": undefined,
      "isOnPlatform": undefined,
      "isTerminus": undefined,
      "mission": "UPAC",
      "noPassenger": undefined,
      "platform": undefined,
      "trainCode": "0929",
    },
    Object {
      "departureTime": "2020-03-10T09:37:00+01:00",
      "displayAttributes": "09:37",
      "displayDestination": "Poissy",
      "isApproaching": undefined,
      "isDeparture": undefined,
      "isOnPlatform": undefined,
      "isTerminus": undefined,
      "mission": "TEDI",
      "noPassenger": undefined,
      "platform": undefined,
      "trainCode": "0937",
    },
    Object {
      "departureTime": "2020-03-10T19:12:00+01:00",
      "displayAttributes": "19:12",
      "displayDestination": "Poissy",
      "isApproaching": undefined,
      "isDeparture": undefined,
      "isOnPlatform": undefined,
      "isTerminus": undefined,
      "mission": "TEDI",
      "noPassenger": undefined,
      "platform": undefined,
      "trainCode": "1912",
    },
    Object {
      "departureTime": "2020-03-11T00:15:00+01:00",
      "displayAttributes": "00:15",
      "displayDestination": "Poissy",
      "isApproaching": undefined,
      "isDeparture": undefined,
      "isOnPlatform": undefined,
      "isTerminus": undefined,
      "mission": "TEDI",
      "noPassenger": undefined,
      "platform": undefined,
      "trainCode": "2415",
    },
  ],
}
`;
